### ISSUE TITLE
attention

### ISSUE THREAD

**Issue by cometta** [huseinzol05/Stock-Prediction-Models#67]
In your Attention class,
 self.X shape is [1 5 1], after executing below 
  x_mean = tf.reduce_mean(self.X, axis = 2)
the shape becomes [1 5], if mean is apply on axis=2 , the values will be the same, and only reduce [1 5 1] to [1 5] . Is this correct?

**Comment by IISuperluminaLII** [comment]
mean over axis 2 on [1 5 1] will be reduced to [1 5] yes
